### PR TITLE
fix: [CDS-49824]: Adding indexes and hint to handle graphql/deployments page

### PR DESCRIPTION
**Pull request by lasdox**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42563)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/execution/ExecutionQueryHelper.java [hunk 1]
```diff
@@ -338,6 +341,24 @@ private void optimizeForService(Query<WorkflowExecution> query, QLIdFilter filte
     utils.setIdFilter(appField, newAppFilter);
   }
 
+  public BasicDBObject getIndexHint(List<QLExecutionFilter> filters) {
+    final List<MongoIndex> wfIndexes = WorkflowExecution.mongoIndexes();
+    for (QLBaseExecutionFilter filter : filters) {
+      if (filter.getEnvironment() != null) {
+        return BasicDBUtils.getIndexObject(wfIndexes, "wfe_executions_search_envIds");
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/graphql/datafetcher/execution/ExecutionQueryHelper.java:348]: We should make these static strings as constant

File: 400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionOptimizationHelper.java [hunk 2]
```diff
@@ -61,27 +64,41 @@ public void enforceAppIdFromChildrenEntities(PageRequest<WorkflowExecution> page
...
+            BasicDBUtils.getIndexObject(WorkflowExecution.mongoIndexes(), "wfe_executions_search_envIds"));
       } else if (WorkflowExecutionKeys.serviceIds.equals(filter.getFieldName())) {
         List<Service> services = hPersistence.createQuery(Service.class)
                                      .filter(ServiceKeys.accountId, accountId)
                                      .field(ServiceKeys.uuid)
                                      .in(asList(filter.getFieldValues()))
                                      .asList();
         services.forEach(service -> appIds.add(service.getAppId()));
+        pageRequest.setIndexHint(
+            BasicDBUtils.getIndexObject(WorkflowExecution.mongoIndexes(), "wfe_executions_search_serviceIds"));
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionOptimizationHelper.java:77]: ditto

**Review by lasdox — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42563)* status: 8 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/graphql/datafetcher/execution/ExecutionQueryHelper.java` line 348 at r2](https://reviewable.io/reviews/harness/harness-core/42563#-NLgjcQEctgPQL_jBlQn-r2-348:-NLvX0q-3Q78GGlqBvfy:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6bb44fd21574eabc1334c26e161a5f8362fe7c58/400-rest/src/main/java/software/wings/graphql/datafetcher/execution/ExecutionQueryHelper.java#L348)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

We should make these static strings as constant
</blockquote></details>

Done.
___
*[`400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionOptimizationHelper.java` line 77 at r2](https://reviewable.io/reviews/harness/harness-core/42563#-NL_zifuatUZRj1PO1Yn-r2-77:-NLvX-WUEfKnaiZjGM68:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6bb44fd21574eabc1334c26e161a5f8362fe7c58/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionOptimizationHelper.java#L77)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42563)* status: 8 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionOptimizationHelper.java` line 77 at r2](https://reviewable.io/reviews/harness/harness-core/42563#-NL_zifuatUZRj1PO1Yn-r2-77:-NLxwIEJ2LdpT6bfumWf:beszo18) ([raw file](https://github.com/harness/harness-core/blob/6bb44fd21574eabc1334c26e161a5f8362fe7c58/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionOptimizationHelper.java#L77)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

still not done

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 3 of 8 files at r1, 1 of 2 files at r3, 2 of 5 files at r4, 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42563)* status: 11 file(s) to review. 1 of 2 approvals obtained.

**Review by hextechpal — CHANGES_REQUESTED** [review]
> Reviewed 2 of 8 files at r1, 1 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42563)* status: 8 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/beans/WorkflowExecution.java` line 98 at r5](https://reviewable.io/reviews/harness/harness-core/42563#-NM2zaj18-1V_n_F-LLn:-NM2zaj18-1V_n_F-LLo:bg902q) ([raw file](https://github.com/harness/harness-core/blob/58cb45322bbaee303106a2a31326fc10f1e8ba4f/400-rest/src/main/java/software/wings/beans/WorkflowExecution.java#L98)):*
> ```Java
>                                           CreatedAtAware, CreatedByAware, KeywordsAware, AccountAccess {
>   public enum IndexName {
>     SEARCH2("search2"),
> ```

why enums can be just static strings

**Review by hextechpal — APPROVED** [review]
> Reviewed 1 of 8 files at r1, 1 of 2 files at r3, 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42563)* status: 7 file(s) to review. 1 of 2 approvals obtained.

**Review by vcalasansh — APPROVED** [review]
> Reviewed 3 of 8 files at r1, 1 of 2 files at r3, 6 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42563)* status: 1 file(s) to review.

**Pull request merged**